src/ODEs/TransientFESpaces.jl

```diff
@@ -0,0 +1,361 @@
+"""
+function Arrays.evaluate(space::AbstractTransientTrialFESpace, t::Nothing)
+  @abstractmethod
+end
+
+Arrays.evaluate(space::FESpace, t::Nothing) = space
+
+"""
+    evaluate!(
+      transient_space::FESpace,
+      space::AbstractTransientTrialFESpace, t::Real
+    ) -> FESpace
+
+Replace the Dirichlet values of the space by those at time `t`.
+"""
+function Arrays.evaluate!(
+  transient_space::FESpace,
+  space::AbstractTransientTrialFESpace, t::Real
+)
+  @abstractmethod
+end
+
+Arrays.evaluate!(transient_space::FESpace, space::FESpace, t::Real) = space
+
+"""
+    (space::AbstractTransientTrialFESpace)(t) -> FESpace
+
+Alias for [`evaluate(space, t)`](@ref).
+"""
+(space::AbstractTransientTrialFESpace)(t) = evaluate(space, t)
+
+# TODO why is this needed?
+@static if VERSION >= v"1.3"
+  (space::FESpace)(t) = evaluate(space, t)
+end
+
+(space::TrialFESpace)(t) = evaluate(space, t)
+(space::ZeroMeanFESpace)(t) = evaluate(space, t)
+
+"""
+    time_derivative(space::AbstractTransientTrialFESpace) -> FESpace
+
+First-order time derivative of the Dirichlet functions.
+"""
+function time_derivative(space::AbstractTransientTrialFESpace)
+  @abstractmethod
+end
+
+# Specialisation for `SingleFieldFESpace`
+function time_derivative(space::SingleFieldFESpace)
+  HomogeneousTrialFESpace(space)
+end
+
+function time_derivative(space::SingleFieldFESpace, ::Val{0})
+  space
+end
+
+function time_derivative(space::SingleFieldFESpace, ::Val)
+  time_derivative(space)
+end
+
+# Specialisation for `MultiFieldFESpace`
+function time_derivative(space::MultiFieldFESpace)
+  MultiFieldFESpace(time_derivative.(space.spaces))
+end
+
+# FESpace interface
+function FESpaces.SparseMatrixAssembler(
+  trial::AbstractTransientTrialFESpace,
+  test::FESpace
+)
+  SparseMatrixAssembler(evaluate(trial, nothing), test)
+end
+
+#########################
+# TransientTrialFESpace #
+#########################
+"""
+    struct TransientTrialFESpace <: AbstractTransientTrialFESpace end
+
+Single-field `FESpace` with transient Dirichlet data.
+"""
+struct TransientTrialFESpace{U,U0} <: AbstractTransientTrialFESpace
+  space::U
+  homogeneous_space::U0
+  transient_dirichlet::Union{Function,Tuple{Vararg{Function}}}
+
+  function TransientTrialFESpace(
+    space::FESpace, transient_dirichlet::Union{Function,Tuple{Vararg{Function}}}
+  )
+    homogeneous_space = HomogeneousTrialFESpace(space)
+    U = typeof(space)
+    U0 = typeof(homogeneous_space)
+    new{U,U0}(space, homogeneous_space, transient_dirichlet)
+  end
+end
+
+# Constructors
+function TransientTrialFESpace(space)
+  HomogeneousTrialFESpace(space)
+end
+
+# AbstractTransientTrialFESpace interface
+function allocate_space(U::TransientTrialFESpace)
+  HomogeneousTrialFESpace(U.space)
+end
+
+Arrays.evaluate(U::TransientTrialFESpace, t::Nothing) = U.homogeneous_space
+
+function Arrays.evaluate!(Ut::FESpace, U::TransientTrialFESpace, t::Real)
+  if U.transient_dirichlet isa AbstractVector
```

Comment:
Hi @AlexandreMagueresse , I have a test failing when multiple BCs are sent to the trial FE space. `U.transient_dirichlet` is either a `Function` of a `Tuple{Vararg{Function}}`, which is not an `AbstractVector`. Then for multiple BCs, it goes to line 166 instead of 164. Could you fix this and add a test for multiple BCs?

Reply:
Hi @oriolcg , thank you for pointing that out. I have just fixed this (using `AbstractVector{<:Function}` instead of `Tuple{Varag{Function}}` to match the `TrialFESpace` constructor. I also added a few tests [here](https://github.com/gridap/Gridap.jl/blob/a337e1fa8a70e313bd9db97bf3594d3fd7d04c8e/test/ODEsTests/TransientFESpacesTests.jl#L42).